src/main/java/edu/byu/cs/util/PhaseUtils.java

```diff
@@ -187,6 +188,19 @@ public static float extraCreditValue(Phase phase) {
         return 0;
     }
 
+    public static CommitVerificationConfig verificationConfig(Phase phase) throws GradingException {
+        var penaltyPct = 10;
+        var forgivenessMinutesHead = 3;
+        return new CommitVerificationConfig(10, 3, 5, penaltyPct, forgivenessMinutesHead);
+        // TODO: Enable this functionality to change the values in next semester.
+        // Don't forget to also update the rubrics in the phase specs and the grade-book.
+//        return switch (phase) {
+//            case Phase0, Phase1 -> new CommitVerificationConfig(8, 2, 5, penaltyPct, forgivenessMinutesHead);
+//            case Phase3, Phase4, Phase5, Phase6 -> new CommitVerificationConfig(12, 3, 5, penaltyPct, forgivenessMinutesHead);
```

Comment:
Should the minimum number of lines changed be a variable as well? Also, would minimum lines, penalty percentage, and forgiveness minutes be better as class constants rather than local magic numbers?

Reply:
I'll move it to a named variable as requested.

In truth, there are a wide range of different formats this code could take; it could quickly become a game of preferences to determine what should be named and what should be a number. Intelli-J inserts the names of the record fields inline when viewing it in an IDE, but this effect obviously won't be observed in the GitHub PR.

None of these are "magic numbers" because they aren't numbers in-lined into code expressions. These are named configuration variables that capable of being adjusted dynamically. The `CommitVerificationConfig` record clearly defines each one and provides information about what they represent. In the functioning code, these are referenced by name. This is the place where they are defined which is consistent with the definition of other fields and strings relating to Phases in this file.

However, this needn't be the place where they are defined. We could just as easily read these numbers out of a configurable table, and the system could handle it. In fact, these numbers have been extracted into this format specifically to simplify and enable the connection to a configurable table in the future. In this way the `CommitVerificationConfig` serves as an interface that must be implemented, but the data can be provided from any of many places.

This is the same reason they won't be replaced with class constants. Class constants cannot be configured remotely, but these values can be configured remotely.